File: frontend/lib/api/conversation.ts
// Conversation API client functions

const API_BASE =
    process.env.NEXT_PUBLIC_API_BASE ||
    process.env.NEXT_PUBLIC_API_URL ||
    "";

export interface ChatMessage {
    role: "user" | "assistant";
    content: string;
}

export interface ChatRequest {
    conversation_id?: string;  // undefined for new conversation
    message: string;
}

export interface ChatResponse {
    conversation_id: string;
    action_type: "ask_more" | "create_task" | "create_project" | "reply" | "review_task" | "confirm_brief" | "update_plan";
    message: string;
    plan?: Record<string, unknown>;
    task?: Record<string, unknown>;  // task data (draft or created)
    planning_session_id?: string;
    stage: string;
    intent?: string;
}

export interface LoginGreetingResponse {
    message: {
        role: "assistant";
        content: string;
        timestamp?: string;
        type?: string;
    };
}

export async function sendChatMessage(
    message: string,
    token: string,
    conversationId?: string
): Promise<ChatResponse> {
    const response = await fetch(`${API_BASE}/conversation/chat`, {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${token}`,
        },
        body: JSON.stringify({
            conversation_id: conversationId,
            message,
        }),
    });

    if (!response.ok) {
        const error = await response.json();
        throw new Error(error.detail || "对话失败");
    }

    return response.json();
}

export async function requestLoginGreeting(token: string): Promise<LoginGreetingResponse> {
    const response = await fetch(`${API_BASE}/conversation/login-greeting`, {
        method: "POST",
        headers: {
            "Content-Type": "application/json",
            Authorization: `Bearer ${token}`,
        },
        body: JSON.stringify({}),
    });

    if (!response.ok) {
        const error = await response.json().catch(() => ({}));
        throw new Error(error.detail || "登录问候生成失败");
    }

    return response.json();
}

File: app/schemas/conversation.py
"""Conversation schemas for multi-turn planning."""
from typing import List, Optional, Dict, Any
from datetime import datetime
from pydantic import BaseModel, Field


class Message(BaseModel):
    """A single message in the conversation."""
    role: str = Field(..., pattern="^(user|assistant)$")
    content: str


class ChatRequest(BaseModel):
    """Request to continue a conversation."""
    conversation_id: Optional[str] = None  # None for new conversation
    message: str = Field(..., min_length=1, max_length=2000)


class ChatResponse(BaseModel):
    """Response from the chat endpoint."""
    conversation_id: str
    action_type: str = Field(..., pattern="^(ask_more|create_task|create_project|reply|review_task|confirm_brief|update_plan)$")
    message: str  # AI's response message
    
    # Optional fields depending on action_type
    plan: Optional[Dict[str, Any]] = None  # For create_project
    task: Optional[Dict[str, Any]] = None  # For create_task
    planning_session_id: Optional[str] = None  # For plan commit/restore
    
    # Metadata
    stage: str  # current conversation stage
    intent: Optional[str] = None
    
    class Config:
        json_schema_extra = {
            "example": {
                "conversation_id": "conv-uuid",
                "action_type": "ask_more",
                "message": "好的！为了帮你制定合理的计划，我需要了解：\n1. 期望多久完成？\n2. 每天能投入多少时间？",
                "stage": "gathering",
                "intent": "complex_project"
            }
        }


class QuickTaskRequest(BaseModel):
    """Request to quickly create a single task."""
    title: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    deadline: datetime
    evidence_type: str = Field(default="none", pattern="^(none|text|number|image)$")


class QuickTaskResponse(BaseModel):
    """Response after creating a quick task."""
    task_id: str
    title: str
    deadline: datetime


class InjectReminderRequest(BaseModel):
    """Request to inject a reminder message into the conversation."""
    data: Dict[str, Any]

